docs/antrea-network-policy.md

```diff
@@ -689,6 +690,61 @@ The following kubectl commands can be used to retrieve CG resources:
     kubectl get cg.core.antrea.tanzu.vmware.com
 ```
 
+## Select Namespace by Name
+
+Kubernetes NetworkPolicies and Antrea-native NetworkPolicies allow selecting
+workloads from Namespaces with the use of a label selector (i.e. `namespaceSelector`).
+However, it is often desired that the Namespaces be selected directly by its `name`
+as opposed to using the `labels` associated with the Namespace. In order to select
+Namespaces by name, Antrea labels Namespaces with a reserved label `antrea.io/metadata.name`,
+which sets it's value to the name of the Namespace. Users can then use this label in the
+`namespaceSelector` field, in both, K8s NetworkPolicies and Antrea-native policies to
+select Namespaces by name. By default, Namespaces are not labeled with reserved name label.
+In order for Antrea controller to label the Namespaces, the `labelsMutator`
+`MutatingWebhookConfiguration` must be enabled. This can be done by generating and applying
+the `antrea.yaml` deployment manifest in the following way:
+
+```bash
+    # From "hack" directory
+    ./generate-manifest.sh --custom-adm-controller
```

Comment:
an alternative solution would be to just ask users to apply an additional yaml manifest, which only includes the `MutatingWebhookConfiguration` resource. It may be a bit easier, as they don't have to clone the repo and run the bash script. However, I don't have a strong preference.

Reply:
I also feel as this is a user doc, maybe just provide the sample YAML here, rather than generating the yaml from the code tree.

Reply:
thanks .. replaced with sample yaml

Reply:
are you'll suggesting we get rid of the change in script and simply ask users to append yaml?

Reply:
I think you can keep the changes to generate-manifest.sh if you want - there is no harm and it may be useful for dev testing?

Reply:
I feel good to keep the option in ./generate-manifest.sh. Might be ok to keep the description of its usage in this doc, besides the sample yaml.

But maybe @antoninbas means to just publish the link of webhook.yml.